compare data output for mcnp.py

qwang252 opened the pull request.

jinneykil committed:
Update 01-nuclide-naming.ipynb

jinneykil committed:
Update 03-nuclear-data.ipynb

jinneykil committed:
Update 01-nuclide-naming.ipynb

scopatz committed:
Merge pull request #788 from kellyrowland/install_instruct

clarify linux source install instructions

Commit:
Merge branch 'develop' of github.com:pyne/pyne into develop

Commit:
Merge branch 'develop' of https://github.com/crivera96/pyne into crivera96-develop

Commit:
changed index.rst 2 to conform to less than 80 char lines.

Commit:
made github visual interface optional; I think this addresses scopatz concerns?

Commit:
Merge branch 'crivera96-develop' into develop

Commit:
Merge pull request #780 from scopatz/ensdf-alpha

hotfix for ENSDF alpha handling

Commit:
Merge branch 'develop' of github.com:pyne/pyne into develop

Commit:
Merge branch 'develop' of https://github.com/pcue/pyne into pcue-develop

Commit:
fixed spacing and character length.

scopatz committed:
Merge pull request #790 from rachelslaybaugh/pcue-develop

Replacement for New_dev_guide #732

Commit:
removed githubsetup folder.

scopatz committed:
Merge pull request #789 from rachelslaybaugh/develop

I think I addressed the issues in Github Installation and Basic Use #733

kellyrowland committed:
fix typo re: HIPAA

scopatz committed:
Merge pull request #791 from kellyrowland/hipaa

fix typo in tutorial re: HIPAA

dschwen committed:
User std::abs (#792)

scopatz committed:
fix EAF data issue

scopatz committed:
made file downloading more robust

scopatz committed:
endl fixes

scopatz committed:
more fixes

dschwen committed:
Regold tests (#793)

scopatz committed:
Merge pull request #793 from dschwen/stdabs_792

User std::abs

dschwen committed:
Add test to check if tutorial files are valid JSON

scopatz committed:
Merge pull request #795 from dschwen/fix_729

Fix PR #729

elliottbiondo committed:
14.04 -> 15.04

elliottbiondo committed:
14.04 -> 15.04 description

Commit:
Changed fortran flags, added ensdf-processing executable copying framework and added ensdf-processing files to cmakelists.txt

Commit:
Added ensdf-processing fortran source, and reference files for tests

Commit:
Added ensdf_processing python interface and tests

Commit:
Added bricc test file reference output

Commit:
Fixed incorrect indentation in setup.py and added nsdflib95.f for ensdf-processing module

Commit:
Removed bricc, and un-needed alphad test file from ensdf-processing module

Commit:
removed bricc test call

Commit:
Added GABS program to ensdf-processing pyne modlue

Commit:
Removed accidental copied line

Commit:
Final pre pr ensdf-processing cleanup

Commit:
Fortran compiler flag changes in cmake/pynemacros.cmake

Commit:
More final cleanup for ensdf_processing pre pr

Commit:
Removed -c flag from unknown fortran compiler

Commit:
Added path check to make sure ensdf executables copied to proper pyne install path.

makeclean committed:
added first version

makeclean committed:
added tests and return from load_data_map

makeclean committed:
add the atomic_data to the amalgamation

makeclean committed:
Added updated version with nicer formatting

Commit:
Added hack to get pyne install path in event of import error when setting pynepath

Commit:
Fixed hack to get ensdf_processing executables copying the first time PyNE is installed, and fixed ensdf_processing docstrings

Commit:
Another attempt at hack to get pyne install directory in setup.py for copying ensdf_processing executables

Commit:
Converted tabs to spaces

Commit:
Fixed a print statement

Commit:
Removed prefix 0 from permission code

Commit:
updated pipe writes to be python3 compadible

Commit:
updated pipe writes to be python3 compadible

makeclean committed:
added python3 compatable string method

makeclean committed:
added comment to say why we set the datapath to nonsense

Commit:
Put in check for type of nuclide vector.

Commit:
Added cram test.

Commit:
Removed unnecessary comment.

Commit:
Removed extraneous print statement.

Commit:
Changed xrange to range

makeclean committed:
added functions to automatically generate atomic_data.cpp & atomic_data.h at install time

makeclean committed:
removed the atomic_data source files such that they are generated at install time

Commit:
Fixed ufunc error.

makeclean committed:
fixed the issue with the src path

makeclean committed:
added last resort option for Conda based people

makeclean committed:
updated for the default amlgamate option

makeclean committed:
updated setup.py to fix silly error

Commit:
Changed pdecay variable to float.

Commit:
Chaned pdecay variable for cram16 test.

Commit:
Changed pdecay to list.

makeclean committed:
Fixed missing file close

makeclean committed:
if . not in path then add src to path fix

makeclean committed:
move files around for consisteny

makeclean committed:
updated documentation for the data  module

Commit:
Merge pull request #807 from mzweig/cram

Added CRAM to alara.py

kellyrowland committed:
change binary download command to conda channel

kellyrowland committed:
make conda binary command instruction more general

kellyrowland committed:
update binary download instructions in readme and docs too

kellyrowland committed:
fix small space typo in binary instruct in docs

katyhuff committed:
Merge pull request #808 from kellyrowland/conda_bin_fix

change binary download command to conda channel

Commit:
Merge pull request #797 from howland/ensdf-processing-alphad

Ensdf processing alphad

Commit:
Added ensdf_processing/delta

Commit:
Added ensdf_processing/gtol tool

Commit:
Added ensdf_processing/hsicc set of tools

Commit:
Fixed gtol doccumentation and added missing endline to test_ensdf_processing

Commit:
removed out of place quotes

makeclean committed:
Addressed review comments

makeclean committed:
renamed code-block type to bash

makeclean committed:
missing new lines

makeclean committed:
more formatting

Commit:
Fixed typo

Commit:
Added interface, tests and reference files for the following ensdf analysis programs: logft, radd, radlst, ruler

Commit:
Added fortran source for logft, radd and ruler ensdf analysis programs

Commit:
Removed print statement

Commit:
Added missing newline to end of file

Commit:
Removed extra newlines

Commit:
Added byte decoding for python3 to radd output

Commit:
Forced reference files to open with ascii encoding

Commit:
Force fortran reference files to be read in with ascii encoding

Commit:
Reveresed codecs change

makeclean committed:
stray OA

elliottbiondo committed:
Merge pull request #806 from makeclean/add_cpp_data

Store Atomic & isotopic data in memory

Commit:
Removed radlist

Commit:
Removed force read encoding as ascii for ensdf test files

Commit:
Removed bldhst

Commit:
Removed old bldhst executable copying code

Commit:
Updated docstrings to be numpydocstring compliant, and changed setup.py ensdf copying from longform to list

Commit:
Updated variable names to be PEP8 compliant, and used a context manager for writing radd output to file

Commit:
Updated setup.py to remove excessive executable copying code, consolidated to array of executable paths

Commit:
Updated setup.py to remove excessive executable copying code, consolidated to array of executable paths

Commit:
Fixed styling issues with documentation

Commit:
Fixed styling issues with documentation

Commit:
Cleanpup and re initializing tests

scopatz committed:
Merge pull request #811 from howland/ensdf-processing-logft

Ensdf processing logft

makeclean committed:
removed unessessary instrinsic function declarations

scopatz committed:
Merge pull request #812 from makeclean/fortran_intrinsics

removed unessessary instrinsic function declarations

Commit:
Merged other ensdf evaluation tools in

Commit:
Added bricc, bldhst and radlist ensdf evaluation tools.  Also expanded briccs interface to include multiple types of use.  Need to fix bricc bug

Commit:
Merge pull request #809 from howland/ensdf-processing-delta

Ensdf processing delta

Commit:
wq
OMerge branch 'develop' of https://github.com/pyne/pyne into ensdf-processing-bricc

Commit:
Updated bldhst test to be python3 compliant and added beginnings of tutorial for ensdf processing

Commit:
Updated ensdf-processing tutorial

ljacobson64 committed:
fix fortran compiler errors on condor

elliottbiondo committed:
Adding capabilitiy of printing photon source totals

elliottbiondo committed:
Automatic printing of e_bounds file

elliottbiondo committed:
Minor errors

elliottbiondo committed:
Forgot to commit file

makeclean committed:
Merge pull request #815 from elliottbiondo/r2s_updates

R2S updates

elliottbiondo committed:
Fixing ipython notebooks

elliottbiondo committed:
Merge branch 'site_fix' of https://github.com/elliottbiondo/pyne into develop

scopatz committed:
Merge pull request #816 from elliottbiondo/site_fix

Fixing ipython notebooks on website

Commit:
Updated bricc tests

Commit:
Reverted temporary change

Commit:
Removed ELE.in & 98AK.. from tutorial folder

Commit:
Merge branch 'develop' of https://github.com/pyne/pyne into ensdf-processing-bricc

elliottbiondo committed:
Build script instructions

elliottbiondo committed:
Merge branch 'develop' of https://github.com/pyne/pyne into doc_update

elliottbiondo committed:
Ubuntu build scripts

scopatz committed:
Merge pull request #818 from elliottbiondo/doc_update

Doc update

qwang252 committed:
add fortran reader to mcnp.py

Commit:
add some fortran specifier to mcnp.py

Commit:
Merge branch 'develop' of https://github.com/pyne/pyne into ensdf-processing-bricc

Commit:
Removed unnessesary parenthases

Commit:
fixed open call to be compliant with py2.x and 3.x

Commit:
New, hopefully faster download stragegy

Commit:
Re triggering travis-ci tests

Commit:
Fixed bricc interface and updated download code

qwang252 committed:
add fortranformat to pyne

ljacobson64 committed:
attempt at fixing travis compile errors

ljacobson64 committed:
consistency

makeclean committed:
Merge pull request #820 from ljacobson64/condor_build

Fix fortran compiler errors when using a source build of GCC

elliottbiondo committed:
doc cleanup

scopatz committed:
Merge pull request #822 from elliottbiondo/doc_fix

doc cleanup

makeclean committed:
fixes the hydrogen issue

scopatz committed:
Merge pull request #825 from makeclean/fix_hydrogen

fixes the hydrogen issue

elliottbiondo committed:
Fixing ALARA matlib error

makeclean committed:
added docker ci for pyne

bizzare typo

makeclean committed:
added alternate way of getting the docker id

elliottbiondo committed:
Merge pull request #829 from makeclean/add_docker_ci

Added Docker CI for PyNE

elliottbiondo committed:
Merge branch 'develop' of https://github.com/pyne/pyne into alara_fix

makeclean committed:
Merge pull request #828 from elliottbiondo/alara_fix

Fixing ALARA matlib error

makeclean committed:
Merge pull request #730 from bprichar/develop

Get cell number from bit array

Commit:
Merge branch 'develop' of https://github.com/pyne/pyne into ensdf-processing-bricc

Commit:
Cleaned up commits, added special case for ensdf_processing to travis-ci script that runs nosetest.

Commit:
Removed travis_wait statement from travis nosetest file

qwang252 committed:
add compare_data.py to compare output results and errors

qwang252 committed:
test1 and test2 contain results and error as columns

gonuke commented:
I'm trying to figure out why it things there are 754 files that are different?

gonuke commented:
Maybe it's all the files from the full development tree of fortranformat.

I think we only need a few files, e.g. the 16 or so files in this file: https://pypi.python.org/packages/source/f/fortranformat/fortranformat-0.2.5.tar.gz#md5=3b8cd134f1c2cb02a7a1119a086ea7f6

available here: https://pypi.python.org/pypi/fortranformat

gonuke commented on pyne/compare_data.py:
```diff
@@ -0,0 +1,37 @@
+   ture if the two columns have the same results otherwise return
+   false. Also return the list of absolute difference btw these 
+   two columns"""
+def compare(mode,file1,file2):
```
I think this function should NOT try to compare files.

It should only compare numbers.

`def compare(val1,err1, val2, err2, mode)`

gonuke replied:
As I've said a couple of times, start very small, and make a function that is very robust and will work for all possible cases of two numbers, each with associated statistical errors.

You are jumping ahead to whole files when we don't need that yet.

qwang252 replied:
yeah, I see, so later yesterday I made a simple comparison btw vals and errs.

qwang252 committed:
revise compare_data.py to unit comparsion

Commit:
putting in a temporary catch for missing ace files

Commit:
Altered ensdf_processing module so that modules requiring a download are not tested by nose suit.  Rather the user has to run an additional setup function.  This is to not slow down overall pyne install and test time

Commit:
Some small corrections to the mesh basics tutorial narrative.

elliottbiondo committed:
Merge pull request #831 from Pshriwise/mesh_basics_edits

Some small corrections to the mesh basics tutorial narrative.

kellyrowland committed:
change *.rst to *.html in style guide to fix broken links

scopatz committed:
Merge pull request #833 from kellyrowland/style_guide_fix

change *.rst to *.html in style guide to fix broken links

Commit:
Split lines over 80 characters in ensdf_processing

Commit:
Fixed more line over 80 char instances, spelling, etc

kellyrowland committed:
Merge pull request #817 from howland/ensdf-processing-bricc

Ensdf processing bricc

rwcarlsen committed:
fix isnan compiler errors for newer clang,gcc

Newer compiler versions (at least clang 3.7.1, gcc 5.3) change their stdlib
shenanigans a bit causing compiler errors like this:

    /home/r/src/cyclus/cyclus/src/pyne.cc:7975:8: error: use of undeclared identifier 'isnan'
      if (!isnan(k_conv)) {

So isnan calls need to be explicitly std:: prefixed - so the pre-existing
macro for apple+win systems was just scoped to include all systems (linux
too).

scopatz committed:
Merge pull request #836 from rwcarlsen/fix-isnan

fix isnan compiler errors for newer clang,gcc

qwang252 committed:
combine mode 2 and 3 and include sigma mode

scopatz committed:
Merge pull request #837 from FlanFlanagan/data_source_fix

putting in a temporary catch for missing ace files

Commit:
added something to deal with a missing xs in cache

Commit:
added some key error catches

yijinhua committed:
Update 00-intro.ipynb

yijinhua committed:
Update 00-intro.ipynb

yijinhua committed:
Update 00-intro.ipynb

yijinhua committed:
Update 00-intro.ipynb

yijinhua committed:
Update 00-intro.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 02-materials.ipynb

yijinhua committed:
Update 04-reaction_names.ipynb

yijinhua committed:
Update 09-outro.ipynb

yijinhua committed:
Update 09-outro.ipynb

yijinhua committed:
Update 09-outro.ipynb

yijinhua committed:
Update 09-outro.ipynb

yijinhua committed:
Update 09-outro.ipynb

Commit:
added key error checking to cross section generation

Commit:
individualized the try excepts for the new keyerror catches

Commit:
fixed try except blocks and added a test for the key error

yijinhua committed:
Update 10-spatial-solver.ipynb

yijinhua committed:
Update 10-spatial-solver.ipynb

yijinhua committed:
Update 09-outro.ipynb

yijinhua committed:
Update 11-ensdf-processing.ipynb

yijinhua committed:
Update 11-ensdf-processing.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 01-nuclide-naming.ipynb

yijinhua committed:
Update 02-materials.ipynb

Commit:
fixed the test structure

scopatz committed:
Merge pull request #841 from FlanFlanagan/data_source_fix

Data source fix

scopatz committed:
accept elements as keys in text based materials

katyhuff committed:
Merge pull request #842 from yijinhua/develop

Documentation Corrections

scopatz committed:
added test for reading in element name from text

FlanFlanagan committed:
Merge pull request #845 from scopatz/iselem

accept elements as keys in text based materials

qwang252 committed:
implement MCNP reader

qwang252 committed:
add mcnp reader with fortran reader

qwang252 committed:
little improvement

Commit:
Merge branch 'develop' of https://github.com/qwang252/pyne into develop

qwang252 committed:
Merge branch 'develop' of https://github.com/qwang252/pyne into develop

qwang252 committed:
Merge branch 'develop' of https://github.com/ngranda/pyne into develop

Commit:
adding a mctal file

qwang252 committed:
Merge branch 'develop' of https://github.com/ngranda/pyne into develop

Commit:
little fixes 06-08

qwang252 committed:
Merge branch 'develop' of https://github.com/ngranda/pyne into develop

qwang252 committed:
merge mcnp_reader with mcnp.py

scopatz committed:
add pointwise cross section access to OpenMC data source

scopatz committed:
a test fix

FlanFlanagan committed:
Merge pull request #850 from scopatz/ptwise

Pointwise interface for OpenMC Data Source

qwang252 committed:
Merge branch 'develop' of https://github.com/pyne/pyne into develop

Commit:
statepoint datasource

Commit:
moved same or none

Commit:
self shielding using pointwise

Commit:
self shielding tests

Commit:
more in line with PEP8

scopatz committed:
Merge pull request #851 from FlanFlanagan/develop

self shielding for group collapse

qwang252 committed:
add tally

Commit:
Merge branch 'develop' of https://github.com/qwang252/pyne into develop

Commit:
update to the mctal class

qwang252 committed:
some changes in tally class

Commit:
small changes to the tally class

qwang252 committed:
add tally dictionary

qwang252 committed:
build kcode dictionary

Commit:
formatting changes

qwang252 committed:
some changes in kcode

qwang252 committed:
Merge branch 'develop' of https://github.com/ngranda/pyne into develop

qwang252 committed:
open the file only once
